### ISSUE TITLE
Feature: Offline backbone caching to bypass FlashAttention requirements and speed up finetuning

### ISSUE THREAD

**Issue by hafnium49** [NVIDIA/Isaac-GR00T#483]
### Problem
Training GR00T N1.6 is currently blocked on hardware that does not support FlashAttention 2. The codebase explicitly enforces this requirement in `gr00t/model/modules/eagle_backbone.py`:

\`\`\`python
# gr00t/model/modules/eagle_backbone.py
assert use_flash_attention, "nvidia/Eagle-Block2A-2B-v2 requires flash attention by default"
\`\`\`

Additionally, `pyproject.toml` imposes a hard dependency on `flash-attn`, which can be difficult to build on certain environments (e.g. non-standard CUDA versions, specific cloud instances, or older GPUs).

Even on supported hardware, the 2B-parameter backbone forward pass creates a massive bottleneck (~5s/step) during frozen-backbone finetuning.

### Proposed Solution: Offline Backbone Caching
I propose an offline caching workflow that separates the backbone inference from the action head training. Since `Gr00tN1d6.forward` already explicitly decouples these steps:

\`\`\`python
# gr00t/model/gr00t_n1d6/gr00t_n1d6.py
backbone_outputs = self.backbone(backbone_inputs)
action_outputs = self.action_head(backbone_outputs, action_inputs)
\`\`\`

We can pre-compute `backbone_outputs` once and stream them directly to `action_head` during training.

### Benefits
1.  **Hardware Compatibility**: Enables training on systems where FlashAttention is unavailable or unstable (including older GPUs and specific server configurations like NVIDIA Grace Blackwell/GX10 in certain modes) by adding an eager-attention fallback path for the feature dumper.
2.  **Throughput**: Increases training speed by **~100x** (from ~5s to <0.05s per step).
3.  **Simplified Dev Loop**: Users can debug and iterate on the action head policy without the VRAM and compute overhead of the 2B VLM.

### Performance Results (SO-101 Dataset)

| Scenario | Time/Step | 50k Steps Total |
|----------|-----------|-----------------|
| Live Backbone | ~5s | ~70 hours |
| **Cached Features** | **<0.05s** | **~40 min** |

### Implementation Overview
I have implemented a complete pipeline available on my fork:
1.  **`dump_features_n16.py`**: A script to extract features with **runtime patching** to bypass the `eagle_backbone.py` FlashAttention assertion.
2.  **`CachedFeatureDataset`**: A dataset class that streams pre-computed embeddings (WebDataset/LMDB).
3.  **`train_cached.py`**: A specialized training loop that loads the frozen backbone but bypasses its forward pass.

## Alternatives
- **TensorRT**: `pyproject.toml` lists optional TRT dependencies, but compiling engines is complex and doesn't solve the "training loop" bottleneck as effectively as caching.
- **Gradient Checkpointing**: Reduces memory but does not solve the compute bottleneck or the hard FlashAttention dependency.

## Additional context
The implementation has been verified on the SO-101 Phase 1 dataset (128k samples).
Reference Branch: **https://github.com/hafnium49/Isaac-GR00T/tree/feature/cached-training**

I believe this would be a valuable addition for the community to enable wider adoption on diverse hardware. Happy to open a PR!
